monix-tail/shared/src/main/scala/monix/tail/Iterant.scala

```diff
@@ -507,6 +507,10 @@ sealed abstract class Iterant[F[_], A] extends Product with Serializable {
   final def doOnFinish(f: Option[Throwable] => F[Unit])(implicit F: Sync[F]): Iterant[F, A] =
     IterantStop.doOnFinish(this, f)(F)
 
+  // @TODO determine the destiny of this method
+  private[tail] final def doOnFullConsumption(f: Option[Throwable] => F[Unit])(implicit F: Sync[F]): Iterant[F, A] =
+    IterantStop.doOnFullConsumption(this, f)(F)
```

Comment:
I added that operation because currently there's no way to handle early stop and full consumption separately (well, without closing over `var`s and checking them, or something like that)

Reply:
Actually this highlights that `doOnFinish` is not correctly defined.

If we are going to use the semantics of `Task.doOnFinish`, since after all that's where I got the name from, then `doOnFinish` shouldn't handle early stop.